region/wreckedship/main/Wrecked Ship Main Shaft.json

```diff
@@ -737,6 +737,108 @@
         "Place the power bomb against the right wall at the top of the stairs between the three obstacles."
       ]
     },
+    {
+      "link": [3, 8],
+      "name": "Temporary Blue (In-Room)",
+      "requires": [
+        "canTemporaryBlue",
+        "canLateralMidAirMorph",
+        {"or": [
+          {"canShineCharge": {
+            "usedTiles": 11,
+            "openEnd": 2,
+            "steepUpTiles": 3
+          }},
+          {"and": [
+            "h_canFrozenEnemyRunway",
+            {"or": [
+              "f_DefeatedPhantoon",
+              {"and": [
+                "SpaceJump",
+                "h_canTrickyFrozenEnemyRunway"
+              ]}
+            ]},
+            {"canShineCharge": {
+              "usedTiles": 13,
+              "openEnd": 1,
+              "steepUpTiles": 3
+            }}
+          ]}
+         ]}
+      ],
+      "clearsObstacles": ["C"],
+      "note": [
+        "Shinecharging up the stairs gives a longer effective runway than shinecharging down them.",
+        "A frozen Atomic (or Covern) can be used for more runway."
+      ],
+      "devNote": "There is 1/2 a unusable tile in this runway."
+    },
+    {
+      "link": [3, 8],
+      "name": "Temporary Blue Chain (In-Room)",
+      "requires": [
+        "canChainTemporaryBlue",
+        "canXRayTurnaround",
+        {"or": [
+          {"and": [
+            "can4HighMidAirMorph",
+            {"doorUnlockedAtNode": 5},
+            {"canShineCharge": {
+              "usedTiles": 11,
+              "openEnd": 2,
+              "steepUpTiles": 7
+            }}
+          ]},
+          {"and": [
+            "h_canFrozenEnemyRunway",
+            {"or": [
+              "f_DefeatedPhantoon",
+              {"and": [
+                "SpaceJump",
+                "h_canTrickyFrozenEnemyRunway"
+              ]}
+            ]},
+            "can4HighMidAirMorph",
+            {"doorUnlockedAtNode": 5},
+            {"canShineCharge": {
+              "usedTiles": 13,
+              "openEnd": 1,
+              "steepUpTiles": 7
+            }}
+          ]},
+          {"canShineCharge": {
```

Comment:
"can4HighMidAirMorph" seems needed for this case too. Same for the next case below.

The runway at node 3 may also be worth considering?: it doesn't require the "can4HighMidAirMorph", and it can also be the longer runway if the door is unlocked at node 3 but not node 5.

Reply:
Bouncing down the stairs makes it so you don't need to do any quick morphs.

The `can4HighMidAirMorph` refers to jump morphing within the door frame.  The runway at 3 is all at door frame height.

I got the shortcharge at door closed 5 faster than open door 3, but 3 has to be a little bit easier.